docs/taxonomy/qna_yaml_best_practices.md

```diff
@@ -0,0 +1,62 @@
+
+- Things to Avoid
+    - Historically, LLM is bad in math
+    - Do not provide complex math calculation in Q&A seeds.
+
+- Context
+    - What if knowledge is based on documents not existing in the base model?
+    - In the qna.yaml file, you can pass context within a chunk of information (text from the document that Q&A are based on). Adding context to the skill QnA file might generate better-quality data.
+
+- Formatting & Front-End specific and may change
+    - How to format data in the Q&A file especially how to format tables?
+    - Currently, only files in Markdown format are supported.
+        - If the files are in any other format, they must be converted to Markdown format
+        - For automatic converters, we recommend experimenting with other Markdown conversions like ‘markdown_strict’, ‘asciidoc’ and ‘gfm’
+
+- Intervene in Training
+    - Can I used generated json files to prompt-tuning (watsonx.ai) or using HuggingFace directly?
+    - The output of SDG is in json format and can also be used for traditional fine-tuning.
+
+- Quantities
+    - The number of seed examples
+    - How many seeds I should provide?
+    - The number of seeds:
+        - Generating ~300 QnA pairs from ~5 seed examples is recommended by InstructLab product team.
+        - Knowledge requires 5 pieces of context from document each with 3 QNAs specific to each context piece for a total of 15 qna pairs.
+        - We tried with less than 300 QnA pairs but found the QnA quality only satisfactory.
+        - The task description should be grounded in the domain/document.
+        - Due to this recommendation we should keep in mind that much complex cases can be splitted into smaller chunks of information
+
+- What is the size limit of context window in the Q&A file (qna.yaml)?
+    - Context size limitation:
+    - There is a ~2300 context size limit in the QnA yaml file.
+    - It is advised to keep the ground truth answers concise to respect this limit.
+
+- After Training
+    - How to check the quality of the data in a large data set of the qna.yaml file?
+    - You don’t have to check out synthetic data generated by the SDG process. After generating synthetic data internally, the IBM Research team is sampling to check quality (no need to check them all, especially for extensive set).
+
+- Quality
+    - How to measure quality of obtained data
```

Comment:
It might be worth mentioning that this is is a qualitative evaluation through chatting chatting with the model.